[Bug] [CI failure]: CUDA illegal memory access during dummy run for stabilityai/stable-audio-open-1.0

### Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>

```text
GPU: 1x NVIDIA L4 (23034 MiB)
NVIDIA driver: 570.133.20
CUDA (driver): 13.0
Python: 3.12
PyTorch: as shipped in the CI image (/usr/local/lib/python3.12/dist-packages/torch)
OS: Linux (CI runner image)

(Full `collect_env.py` output to be attached; the failure was reproduced inside the
project CI runner that executes `tests/e2e/online_serving/test_stable_audio_open.py`.)
```

</details>


### Your code version

<details>
<summary>The commit id or version of vllm</summary>

```text
vLLM server version 0.20.0 (as printed by the CI logs)
```
</details>
<details>
<summary>The commit id or version of vllm-omni</summary>

```text

```
</details>


### 🐛 Describe the bug

Please check the CI failure: https://buildkite.com/vllm/vllm-omni/builds/8772/canvas?sid=019def09-524a-4441-b3a9-19847aa3846e&tab=output

When launching the OmniServer with the Stable Audio Open pipeline, the dummy / warm-up
run crashes with `CUDA error: an illegal memory access was encountered` while Triton is
autotuning a fused `add + native_layer_norm` kernel that was generated by `torch.compile`
for a transformer block wrapped by the TeaCache hook.

Command that triggers it (from the CI log):

```sh
/usr/bin/python3 -m vllm_omni.entrypoints.cli.main serve \
    stabilityai/stable-audio-open-1.0 \
    --omni --quantization fp8 --stage-init-timeout 600 --init-timeout 900
```

Sequence of events from the log:

1. Model loads successfully (`Model loading took 2.7891 GiB and 51.83 s`).
2. `transformer compiled with torch.compile` and `TeaCache applied with rel_l1_thresh=0.2,
   transformer_class=StableAudioDiTModel`.
3. During `dummy run to warm up the model`, Triton emits:
   ```
   [rank0]:W0503 18:19:55 torch/_inductor/utils.py:1731] [0/0_1]
       Not enough SMs to use max_autotune_gemm mode
   ```
4. Autotune for `triton_red_fused__unsafe_view_add_native_layer_norm_5` then crashes:

```
ERROR diffusion_worker.py:920 Error executing method 'execute_model'.
  File "/workdir/vllm_omni/diffusion/worker/diffusion_model_runner.py", line 276, in execute_model
    output = self.pipeline.forward(req)
  File "/workdir/vllm_omni/diffusion/models/stable_audio/pipeline_stable_audio.py", line 548, in forward
    noise_pred = self.transformer(...)
  File "/workdir/vllm_omni/diffusion/hooks/base.py", line 273, in dispatch
    output = self._new_fwd_impl_hook.new_forward(self.module, *args, **kwargs)
  File "/workdir/vllm_omni/diffusion/cache/teacache/hook.py", line 168, in new_forward
    outputs = ctx.run_transformer_blocks()
  File "/workdir/vllm_omni/diffusion/cache/teacache/extractors.py", line 896, in run_transformer_blocks
    h = block(...)
  ...
  File "/tmp/torchinductor_root/3l/c3lpvxj5opcm7m57mgbsreg73sd5nvntwriisrdjvu3mchqvhgen.py", line 900, in call
    triton_red_fused__unsafe_view_add_native_layer_norm_5.run(
        arg4_1, buf16, buf33, arg20_1, arg21_1, buf37,
        triton_red_fused__unsafe_view_add_native_layer_norm_5_xnumel, 1536, stream=stream0)
  File ".../torch/_inductor/runtime/triton_heuristics.py", line 1176, in autotune_to_one_config
  File ".../torch/_inductor/runtime/benchmarking.py", line 376, in benchmark_gpu
    torch.cuda.synchronize()
torch.AcceleratorError: CUDA error: an illegal memory access was encountered
```

This propagates up and aborts orchestrator startup:

```
RuntimeError: Dummy run failed: CUDA error: an illegal memory access was encountered
  File "/workdir/vllm_omni/diffusion/diffusion_engine.py", line 447, in _dummy_run
    raise RuntimeError(f"Dummy run failed: {output.error}")
```

#### Expected behavior

Dummy run on L4 (Ada, SM 8.9, 24 SMs) should succeed for `stable-audio-open-1.0` with the
default TeaCache + `torch.compile` configuration, or vllm-omni should automatically fall
back to a safer Inductor configuration when `Not enough SMs to use max_autotune_gemm mode`
is detected.

#### Suspected root cause

* The combination of TeaCache wrapping (`diffusion/cache/teacache/hook.py:168` →
  `extractors.py:896`) + `torch.compile` of the transformer block + Inductor `max_autotune`
  picks an autotune config whose shared-memory / index assumptions are invalid on a
  small-SM GPU (L4), causing `cudaErrorIllegalAddress` from a Triton kernel.
* The first TeaCache step takes the "full forward" path, which can present input
  shapes/strides that differ from what `torch._dynamo` cached, increasing the chance of
  hitting an Inductor edge case.

#### Suggested next steps for the maintainers

* On GPUs where Inductor reports `Not enough SMs to use max_autotune_gemm mode`,
  automatically disable `max-autotune` (or set `mode="reduce-overhead"` / `"default"`)
  in the `torch.compile` call inside `diffusion_model_runner.py` for the Stable Audio
  pipeline.
* Optionally skip `torch.compile` when TeaCache is enabled, since the cache hook already
  changes the per-step execution graph.
* Reproduce locally with `CUDA_LAUNCH_BLOCKING=1 TORCH_USE_CUDA_DSA=1
  TORCHINDUCTOR_MAX_AUTOTUNE=0` to confirm autotune is the trigger.

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://vllm-omni.readthedocs.io), which can answer lots of frequently asked questions.

--- a/tests/diffusion/attention/test_flash_attn.py
+++ b/tests/diffusion/attention/test_flash_attn.py
@@ -15,9 +15,12 @@ import torch
 from vllm_omni.diffusion.attention.backends.abstract import AttentionMetadata
 from vllm_omni.diffusion.attention.backends.flash_attn import FlashAttentionImpl
 from vllm_omni.diffusion.attention.backends.sdpa import SDPAImpl
+from vllm_omni.diffusion.attention.backends.utils import fa  # noqa: E402
 from vllm_omni.platforms import current_omni_platform
 
 is_gpu = current_omni_platform.is_cuda_alike() or current_omni_platform.is_xpu()
+HAS_FLASH_ATTN = fa.HAS_FLASH_ATTN
+flash_attn_func = fa.flash_attn_func  # noqa: N813
 
 
 def create_attention_mask(batch_size: int, seq_len: int, valid_len: int, device: torch.device) -> torch.Tensor:
@@ -262,6 +265,36 @@ def test_fa_vs_sdpa():
     assert mean_diff < 0.001, f"Mean difference {mean_diff} exceeds threshold 0.001"
 
     print("✓ Case 2 PASSED: FA and SDPA outputs are very close!")
+
+
+@pytest.mark.skipif(not is_gpu, reason="FlashAttention requires CUDA or XPU")
+def test_flash_attn_func_preferred_over_varlen():
+    """Test flash_attn_func availability and basic forward call."""
+    if not HAS_FLASH_ATTN:
+        pytest.skip("No Flash Attention available")
+
+    if flash_attn_func is None:
+        pytest.skip("flash_attn_func not available, will use varlen fallback")
+
+    device = torch.device(current_omni_platform.device_type)
+    dtype = torch.bfloat16
+
+    num_heads, head_dim = 8, 64
+    fa_impl = FlashAttentionImpl(
+        num_heads=num_heads, head_size=head_dim, softmax_scale=1.0 / (head_dim**0.5), causal=False
+    )
+
+    torch.manual_seed(42)
+    q = torch.randn(1, 32, num_heads, head_dim, device=device, dtype=dtype)
+    k = q.clone()
+    v = q.clone()
+
+    attn_metadata = AttentionMetadata(attn_mask=None)
+    output = fa_impl.forward(q, k, v, attn_metadata)
+
+    assert output.shape == q.shape
+    assert not torch.isnan(output).any()
+    print("✓ flash_attn_func forward works correctly!")
 
 
 if __name__ == "__main__":

--- a/vllm_omni/diffusion/attention/backends/flash_attn.py
+++ b/vllm_omni/diffusion/attention/backends/flash_attn.py
@@ -140,6 +140,7 @@ class FlashAttentionImpl(AttentionImpl):
         """CUDA/ROCm/MUSA flash attention implementation."""
         from vllm_omni.diffusion.attention.backends.utils.fa import (
             HAS_FLASH_ATTN,
+            flash_attn_func,
         )
 
         if not HAS_FLASH_ATTN:
@@ -158,6 +159,16 @@ class FlashAttentionImpl(AttentionImpl):
                 value,
                 attention_mask,
             )
+
+        if flash_attn_func is not None:
+            out = flash_attn_func(
+                query,
+                key,
+                value,
+                causal=self.causal,
+                softmax_scale=self.softmax_scale,
+            )
+            return self._unwrap_flash_output(out)
 
         return self._forward_varlen_dense(
             query,

--- a/vllm_omni/diffusion/attention/backends/utils/fa.py
+++ b/vllm_omni/diffusion/attention/backends/utils/fa.py
@@ -51,27 +51,28 @@ else:
     # CUDA: try FA3 -> FA2 fallback chain
     # Try FA3 from fa3-fwd PyPI package
     try:
-        from fa3_fwd_interface import flash_attn_varlen_func  # noqa: F401
+        from fa3_fwd_interface import flash_attn_func, flash_attn_varlen_func  # noqa: F401
     except (ImportError, ModuleNotFoundError):
         pass
 
     # Fallback: Try FA3 from flash-attention source build
-    if flash_attn_varlen_func is None:
+    if flash_attn_func is None:
         try:
-            from flash_attn_interface import flash_attn_varlen_func  # noqa: F401
+            from flash_attn_interface import flash_attn_func, flash_attn_varlen_func  # noqa: F401
         except (ImportError, ModuleNotFoundError):
             pass
 
     # Fallback: Try FA2 from flash-attn package (try multiple import paths)
-    if flash_attn_varlen_func is None:
+    if flash_attn_func is None:
         try:
-            from flash_attn import flash_attn_varlen_func  # noqa: F401
+            from flash_attn import flash_attn_func, flash_attn_varlen_func  # noqa: F401
         except (ImportError, ModuleNotFoundError):
             pass
 
-    if flash_attn_varlen_func is None:
+    if flash_attn_func is None:
         try:
             from flash_attn.flash_attn_interface import (  # noqa: F401
+                flash_attn_func,
                 flash_attn_varlen_func,
             )
         except (ImportError, ModuleNotFoundError):

